python/llm/example/GPU/QLoRA-FineTuning/qlora_finetuning.py

```diff
@@ -43,13 +42,16 @@
     data = load_dataset(dataset_path)
     data = data.map(lambda samples: tokenizer(samples["quote"]), batched=True)
     model = AutoModelForCausalLM.from_pretrained(model_path,
-                                                load_in_low_bit="nf4",
+                                                load_in_low_bit="sym_int4",
+                                                modules_to_not_convert=["lm_head"],
                                                 optimize_model=False,
                                                 torch_dtype=torch.float16,
-                                                modules_to_not_convert=["lm_head"],)
-    model = model.to('xpu')
-    model.gradient_checkpointing_enable()
-    model = prepare_model_for_kbit_training(model)
+                                                )
+    model = model.to("cpu")
+    # model.gradient_checkpointing_enable()
+    model = prepare_model_for_kbit_training(model, use_gradient_checkpointing=False)
+
+    model.enable_input_require_grads()
```

Comment:
Somehow both setting `use_gradient_checkpoint=False` and calling `model.enable_input_require_grads()` is necessary. I still have not figure out why.